magefile.go

```diff
@@ -294,6 +294,54 @@ func (Build) GenerateConfig() error {
 	return sh.Copy(filepath.Join(buildDir, configFile), filepath.Join(metaDir, configFile))
 }
 
+// WindowsArchiveRootBinary compiles a binary to be placed at the root of the windows elastic-agent archive. This binary
+// is a thin proxy to the actual elastic-agent binary that resides in the data/elastic-agent-{commit-short-sha}
+// directory of the archive.
+func (Build) WindowsArchiveRootBinary() error {
```

Comment:
To ease future development and testing, we could make it possible to build this for the other OSes. There's no reason it wouldn't work, it's just unnecessary. The job object APIs are already stubbed.

The archive binary is larger than a symlink so I'm not sure we want to use it by default, but perhaps a flag to switch to this so Windows can be taken out of the primary edit code/develop loop and only used for final testing. Dealing with Windows on Mac is a pain currently.

Reply:
that's the reason I didn't add a windows tag in the main.go of this wrapper subpackage. I can compile that from mac and linux, since this has CGO disabled, it compiles just fine from any OS but it targets Windows only. we want something more out of it? 🤔

Reply:
You should be able to run this executable on non-Windows. We could allow building an agent package for Linux/Mac with the proxy binary but I'm fine just doing that with copy paste.

This is something we can address as a follow up, I don't want to block progress here with changes to this. I just want to highlight there is no actual cross-platform dependency here, besides our choice to use the proxy binary only on Windows.